### ISSUE TITLE
oxfmt: non-idempotent formatting of HTML comments inside markdown list items

### ISSUE THREAD

**Issue by SBoudrias** [oxc-project/oxc#21314]
## Description

oxfmt is not idempotent when a markdown file contains an HTML comment (`<!-- ... -->`) that immediately follows a list item (as a list continuation). Each format pass increases the comment's indentation by 2 spaces, so the output never converges.

## Minimal reproduction

Input file:

```markdown
- List item
<!-- html comment -->
```

After **pass 1**:

```markdown
- List item
  <!-- html comment -->
```

After **pass 2**:

```markdown
- List item
    <!-- html comment -->
```

After **pass 3**:

```markdown
- List item
      <!-- html comment -->
```

The indentation grows by 2 spaces on every run indefinitely.

## Expected behavior

The formatter should be idempotent: running it a second time on already-formatted output should produce no changes.

## Actual behavior

The comment's indentation increases by 2 spaces on every pass.

## Impact

In our CI setup, `oxfmt` runs in write mode and commits any changes back to the PR branch. Because the output is never stable, this triggers a new lint fix commit on every pipeline run, creating an infinite loop that exhausts CI resources.

## Version

oxfmt 0.44.0

## Workaround

Moving the comment before the list item (at 0-indent, making it a top-level HTML block rather than a list continuation) produces stable output:

```markdown
<!-- html comment -->

- List item
```

**Comment by leaysgur** [comment]
Currently, Oxfmt simply delegates the formatting of Markdown files to Prettier, and there is nothing that can be done for now.

It may already exist, but please report this as an issue to the upstream repository.

**Comment by SBoudrias** [comment]
I don't encounter this issue with Prettier. This makes me feel it comes from interaction between the 2 tools.

**Comment by leaysgur** [comment]
OK, then please create a minimum reproduction?

```
❯ cat a.md
- List item
<!-- html comment -->

❯ npx oxfmt@0.44.0 a.md
Finished in 55ms on 1 files using 8 threads.

❯ cat a.md
- List item
<!-- html comment -->
```
